pyfar/classes/audio.py

```diff
@@ -1588,14 +1602,19 @@ def _get_arithmetic_data(data, domain, cshape, matmul, audio_type):
         Signal. `np.asarray(data)` otherwise.
     """
     if isinstance(data, (Signal, TimeData, FrequencyData)):
+        data_ = data.copy()
+        # check if complex casting of any input signal is necessarry
+        if complex:
```

Comment:
Could this be simplified to
```suggestion
        if isinstance(data_, (Signal, TimeData)):
           data_.complex = complex
```
We could probably do this to improve readability whithout worring that the complex setter might be called even if it is already set correctly.